src/transformState.js

```
@@ -5,7 +5,30 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const remove of action.keysToRemove) {
+          delete state[remove];
+        }
+        break;
+
+      case 'clear':
+        for (const clear in state) {
+          delete state[clear];
+        }
+        break;
+
+      default:
+        break;
```

Let's throw an error here instead.
Read more - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw